overviews/core/_posts/2012-09-20-futures.md

```diff
@@ -36,6 +36,134 @@ in the case of an application which requires blocking futures, for an underlying
+
+The `scala.concurrent` package comes out of the box with an `ExecutionContext` implementation, a global static thread pool.
+It is also possible to adapt an `Executor` into an `ExecutionContext`, and you are of course free to implement your own. 
+
+
+### The Global Execution Context
+
+`ExecutionContext.global` is an `ExecutionContext` backed by a [ForkJoinPool](http://docs.oracle.com/javase/tutorial/essential/concurrency/forkjoin.html). It should be sufficient for most situations but requires some care.  
+A `ForkJoinPool` manages a limited amount of threads (the maximum amount of thread being refered to as parallelism level). That limit can be exceeded in the presence of blocking computation (the pool must be notified though). 
+
+By default the `ExecutionContext.global` set the parallelism level of its underlying fork-join pool to the amount of available processors ([Runtime.availableProcessors](http://docs.oracle.com/javase/7/docs/api/java/lang/Runtime.html#availableProcessors%28%29)).
+This configuration can be overriden by setting one (or more) of the following VM attribute:
+
+  * scala.concurrent.context.minThreads - defaults to `Runtime.availableProcessors`
+  * scala.concurrent.context.numThreads - can be a number or a multiplier (N) in the form 'xN' ;  defaults to `Runtime.availableProcessors`
+  * scala.concurrent.context.maxThreads - defaults to `Runtime.availableProcessors`
+
+The parallelism level will be set to `numThreads` as long as it remains within `[minThreads; maxThreads]`.
+
+As stated above the `ForkJoinPool` can increase the amount of threads beyond its `parallelismLevel` in the presence of blocking computation. As explained in the `ForkJoinPool` API, this is only possible if the pool is explicitly notified:
+
+    import scala.concurrent.Future
+    import scala.concurrent.forkjoin._
+
+    // the following is equivalent to `implicit val ec = ExecutionContext.global`
+    import ExecutionContext.Implicits.global
+
+    Future {
+      ForkJoinPool.managedBlock(
+        new ManagedBlocker {
+           var done = false
+
+           def block(): Boolean = {
+             try {
+               myLock.lock()
+               // ...
+             } finally {
+              done = true
+             }
+             true
+           }
+
+           def isReleasable: Boolean = done
+        }
+      )
+    }
+
+
+Fortunately the concurrent package provides a convenient way for doing so:
+
+    import scala.concurrent.Future
+	  import scala.concurrent.blocking
+
+    Future {
+      blocking {
+        myLock.lock()
+        // ...
+      }
+    }
+
+Note that `blocking` is a general construct that will be discussed more in depth [below](#in_a_future).
+
+Last but not least, you must remember that the `ForkJoinPool` isn't designed for long lasting blocking operations. Even when notified with `blocking` the pool might not spawn new workers as you would expect, and when new workers are created they can be as many as 32767. 
```

Comment:
It might be helpful to point out that `blocking` provides an opportunity to the underlying thread pool to spawn spare workers *temporarily*, i.e., spare workers terminate when possible. Moreover, the behavior is typically much better-behaved than with, e.g., a JDK `Executors.newCachedThreadPool()`.